apis/topology/v1beta1/topology_types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// TopologySpec defines the desired state of Topology
+type TopologySpec struct {
+	// +kubebuilder:validation:Optional
+	// TopologySpreadConstraints exposes topologySpreadConstraints that are
+	// applied to the StatefulSet
+	TopologySpreadConstraints *[]corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+
+	// Affinity exposes PodAffinity, PodAntiaffinity and NodeAffinity overrides
+	// that are applied to StatefulSet/Deployments
+	// +optional
```

Comment:
nit: probably OK as is but in the future would be nice to use `// +kubebuilder:validation:Optional` consistently

Reply:
what's the suggestion here? remove `// +optional` ?

Reply:
Removed `//+optional`

Reply:
The suggestion was that when a field need to be optional use `// +kubebuilder:validation:Optional` as the tag instead of `//+optional` to make the usage of these tags with similar meaning consistent.